pkg/agent/proxy/endpoints.go

```diff
@@ -95,10 +109,42 @@ func (t *endpointsChangesTracker) OnEndpointUpdate(previous, current *corev1.End
 	return len(t.changes) > 0
 }
 
+// EndpointSliceUpdate updates given the service's endpoints change map based on the <previous, current> endpoints pair.
+// It returns true if items changed, otherwise it returns false. Will add/update/delete items of endpointsChange Map.
+// If removeSlice is true, slice will be removed, otherwise it will be added or updated.
+func (t *endpointsChangesTracker) OnEndpointSliceUpdate(endpointSlice *discovery.EndpointSlice, removeSlice bool) bool {
+	if _, has := supportedEndpointSliceAddressTypes[string(endpointSlice.AddressType)]; !has {
+		klog.V(4).Infof("EndpointSlice address type not supported: %s", endpointSlice.AddressType)
+		return false
+	}
+
+	// This should never happen
+	if endpointSlice == nil {
```

Comment:
If so, do we need to check this? and if it's really nil, I believe L116 already panic because of the usage of `endpointSlice.AddressType`

Reply:
Thanks, I moved the nil checking to the beginning of this method.